core/src/main/java/org/mobicents/cluster/DefaultMobicentsCluster.java

```diff
@@ -215,26 +214,38 @@ public void run() {
 		
 	}
 	
-	@BuddyGroupChanged
-	public void onBuddyGroupChangedEvent(BuddyGroupChangedEvent event) {
-		//here we only update stored information, it happens after view change, so take over should be taken care off.
-		Node root = event.getCache().getRoot();
-		root.put(BUDDIES_STORE, event.getBuddyGroup().getBuddies());
-		
+	
+	
+	@SuppressWarnings("rawtypes")
+	@CacheEntryRemoved
+	public void cacheEntryRemoved(CacheEntryRemovedEvent event){
+		if (logger.isDebugEnabled()) {
+			logger.debug("cacheEntryRemoved : event[ "+ event +"]");
+		}
+		if(!event.isPre() && !event.isOriginLocal() && event.getKey() != null && (event.getKey() instanceof NodeKey)  && ((NodeKey)event.getKey()).getContents() == Type.STRUCTURE){
+			
+			Fqn changed = ((NodeKey)event.getKey()).getFqn();
+			
+			final DataRemovalListener dataRemovalListener = dataRemovalListeners.get(changed.getParent());
+			if (dataRemovalListener != null) {
+				dataRemovalListener.dataRemoved(changed);
```

Comment:
When a node removed from the tree cache (a node now represents a timer) and the event is originated from another node we should trigger the dataRemovalListener in order the check whether the local timer has to be stopped. See the 2nd test case.